src/reducers/console.js

```diff
@@ -12,20 +10,20 @@ export default function console(stateIn, {type, payload, meta}) {
 
   switch (type) {
     case 'CONSOLE_VALUE_PRODUCED':
-      return state.update(
-        payload.key,
+      return state.updateIn(
+        ['history', payload.key],
         input => input.set(
           'value',
           payload.value,
         ).set(
           'evaluatedByCompiledProjectKey',
           payload.compiledProjectKey,
         ),
-      );
-
+      ).delete('currentInputValue').
+        delete('nextConsoleEntry').
+        delete('historyEntryIndex');
     case 'CONSOLE_ERROR_PRODUCED':
-      return state.update(
-        payload.key,
+      return state.updateIn(['history', payload.key],
```

Comment:
Minor thing, but project style is to have the array on its own line (basically it’s all-or-nothing in terms of multiline arguments)

Reply:
Makes sense